Use 0x API

romanbarbo opened the pull request.

romanbarbo committed:
Use 0x API

romanbarbo committed:
Remove wido package

romanbarbo committed:
Set WIDO_FEE_BPS

jainkunal commented on src/wido-compound-sdk.ts:
```diff
@@ -188,25 +191,18 @@ export class WidoCompoundSdk {
       : "0";
-    const minToAmount = supported && quoteResponse.minToTokenAmount
-      ? quoteResponse.minToTokenAmount
+    const minToAmount = minToTokenAmount
```
Let's raise an error when the minToTokenAmount is not present instead of setting it to 0

jainkunal commented on src/wido-compound-sdk.ts:
```diff
@@ -272,20 +268,12 @@ export class WidoCompoundSdk {
       sigs,
-      widoSwap,
-      cometAddress
+      swapQuote.data
```
It seems like we are passing calldata for 0x when the contract expects it to be WidoRouter calldata. Wrap the 0x calldata in WidoRouter's executeOrder function.

jainkunal commented on src/wido-compound-sdk.ts:
```diff
@@ -217,7 +213,7 @@ export class WidoCompoundSdk {
     // construct & return quote
     return {
-      isSupported: supported,
+      isSupported: true,
```
Remove `to` and `tokenManager` fields since they are not used.

romanbarbo committed:
Remove unused 'to' and 'tokenManager' fields

romanbarbo committed:
Add CollateralSwap addresses, remove LoanProviders

Removed the getBestProvider method and the LoanProviders class because both were unused

romanbarbo committed:
Add zeroExApiKey param to WidoCompoundSdk

romanbarbo committed:
Generate calldata for wido router order execution

romanbarbo committed:
Fix test helpers

romanbarbo committed:
Update IWidoCollateralSwap_ABI & minor fixes

romanbarbo committed:
Fix minToAmount and fee calculation

romanbarbo committed:
Bump version

romanbarbo committed:
Add Retry to contract calls
